From abd8555549e48367b9edb9baaa61b40afcfd6231 Mon Sep 17 00:00:00 2001
From: Leonard Foerster <foersleo@amazon.com>
Date: Tue, 25 Jul 2023 14:27:14 +0000
Subject: [PATCH] linux-firmware: usb: remove firmware for USB/Serial/PCMCIA
 devices

Bottlerocket does not configure most drivers for USB,PCMCIA, and serial devices
for any of its kernels. Without those drivers, the firmware for these
devices is not of use and does not need to be shipped by us.

The following list maps the driver names as specified in WHENCE to
kernel config options enabling drivers. That way we are able to find the
firmware for any driver we may be enabling in the future.

* kaweth - CONFIG_USB_KAWETH
* keyspan - CONFIG_SERIAL_KEYSPAN
* keyspan_pda - CONFIG_SERIAL_KEYSPAN_PDA
* emi26 - CONFIG_USB_EMI26
* emi62 - CONFIG_USB_EMI62
* ti_usb_3410_5052 - CONFIG_USB_SERIAL_TI

Signed-off-by: Leonard Foerster <foersleo@amazon.com>
---
diff -ruN usb-step1/LICENCE.agere usb-step2/LICENCE.agere
--- usb-step1/LICENCE.agere	2025-11-18 03:34:08.610979351 +0000
+++ usb-step2/LICENCE.agere	1970-01-01 00:00:00.000000000 +0000
@@ -1,77 +0,0 @@
-agere_sta_fw.bin -- 9.48 Hermes I
-agere_ap_fw.bin  -- 9.48 Hermes I
-
-The above firmware images were compiled from the Agere linux driver
-wl_lkm_718_release.tar.gz, and dumped. The driver is coverred by the
-following copyright and software license.
-
- * SOFTWARE LICENSE
- *
- * This software is provided subject to the following terms and conditions,
- * which you should read carefully before using the software.  Using this
- * software indicates your acceptance of these terms and conditions.  If you do
- * not agree with these terms and conditions, do not use the software.
- *
- * COPYRIGHT © 1994 - 1995  by AT&T.                All Rights Reserved
- * COPYRIGHT © 1996 - 2000 by Lucent Technologies.  All Rights Reserved
- * COPYRIGHT © 2001 - 2004  by Agere Systems Inc.   All Rights Reserved
- * All rights reserved.
- *
- * Redistribution and use in source or binary forms, with or without
- * modifications, are permitted provided that the following conditions are met:
- *
- * . Redistributions of source code must retain the above copyright notice, this
- *    list of conditions and the following Disclaimer as comments in the code as
- *    well as in the documentation and/or other materials provided with the
- *    distribution.
- *
- * . Redistributions in binary form must reproduce the above copyright notice,
- *    this list of conditions and the following Disclaimer in the documentation
- *    and/or other materials provided with the distribution.
- *
- * . Neither the name of Agere Systems Inc. nor the names of the contributors
- *    may be used to endorse or promote products derived from this software
- *    without specific prior written permission.
- *
- * Disclaimer
- *
- * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
- * INCLUDING, BUT NOT LIMITED TO, INFRINGEMENT AND THE IMPLIED WARRANTIES OF
- * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.  ANY
- * USE, MODIFICATION OR DISTRIBUTION OF THIS SOFTWARE IS SOLELY AT THE USERS OWN
- * RISK. IN NO EVENT SHALL AGERE SYSTEMS INC. OR CONTRIBUTORS BE LIABLE FOR ANY
- * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
- * ON ANY THEORY OF LIABILITY, INCLUDING, BUT NOT LIMITED TO, CONTRACT, STRICT
- * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
- * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
- * DAMAGE.
-
-The following statement from Agere clarifies the status of the firmware
-
----
-I would like to confirm that the two drivers; Linux LKM Wireless Driver
-Source Code, Version 7.18 and Linux LKM Wireless Driver Source Code,
-Version 7.22 comply with Open Source BSD License. Therefore the source
-code can be distributed in unmodified or modified form consistent with
-the terms of the license.
-
-The Linux driver architecture was based on two modules, the MSF (Module
-specific functions) and the HCF (Hardware Control Functions).  Included
-in the HCF is run-time firmware (binary format) which is downloaded into
-the RAM of the Hermes 1/2/2.5 WMAC.
-
-This hex coded firmware is not based on any open source software and
-hence it is not subject to any Open Source License.  The firmware was
-developed by Agere and runs on the DISC processor embedded within the
-Hermes 1/2/2.5 Wireless MAC devices. 
-
-Hope this helps.
-
-Sincerely,
-
-Viren Pathare
-Intellectual Property Licensing Manager
-Agere
----
diff -ruN usb-step1/LICENCE.ene_firmware usb-step2/LICENCE.ene_firmware
--- usb-step1/LICENCE.ene_firmware	2025-11-18 03:34:08.612636944 +0000
+++ usb-step2/LICENCE.ene_firmware	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-copyright (c) 2011, ENE TECHNOLOGY INC.
-
-Permission to use, copy, modify, and/or distribute this software for any purpose
-with or without fee is hereby granted, provided that the above copyright notice
-and this permission notice appear in all copies.
-
-THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
-WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL
-IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT
-SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
-CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
-LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
-NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
-WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
diff -ruN usb-step1/LICENCE.kaweth usb-step2/LICENCE.kaweth
--- usb-step1/LICENCE.kaweth	2025-11-18 03:34:08.612636944 +0000
+++ usb-step2/LICENCE.kaweth	1970-01-01 00:00:00.000000000 +0000
@@ -1,28 +0,0 @@
-Copyright 1999 Kawasaki LSI.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions
-are met:
-1. Redistributions of source code must retain the above copyright
-   notice, this list of conditions and the following disclaimer.
-2. Redistributions in binary form must reproduce the above copyright
-   notice, this list of conditions and the following disclaimer in the
-   documentation and/or other materials provided with the distribution.
-3. All advertising materials mentioning features or use of this software
-   must display the following acknowledgement:
-       This product includes software developed by Kawasaki LSI.
-4. Neither the name of the company nor the names of its contributors
-   may be used to endorse or promote products derived from this software
-   without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY KAWASAKI LSI ``AS IS'' AND ANY EXPRESS OR
-IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED.  IN NO EVENT SHALL KAWASAKI LSI BE LIABLE FOR ANY DIRECT,
-INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
-HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
-STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
-IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-POSSIBILITY OF SUCH DAMAGE.
diff -ruN usb-step1/LICENCE.moxa usb-step2/LICENCE.moxa
--- usb-step1/LICENCE.moxa	2025-11-18 03:34:08.613636947 +0000
+++ usb-step2/LICENCE.moxa	1970-01-01 00:00:00.000000000 +0000
@@ -1,16 +0,0 @@
-The software accompanying this license statement (the “Software”)
-is the property of Moxa Inc. (the “Moxa”), and is protected by
-United States and International Copyright Laws and International
-treaty provisions. No ownership rights are granted by this
-Agreement or possession of the Software. Therefore, you must treat
-the Licensed Software like any other copyrighted material. Your
-rights and obligations in its use are described as follows:
-
-1. You may freely redistribute this software under this license.
-2. You may freely download and use this software on Moxa's device.
-3. You may not modify or attempt to reverse engineer the software, or
-   make any attempt to change or even examine the source code of the
-   software.
-4. You may not re-license or sub-license the software to any person or
-   business, using any other license.
-5. Moxa(r) is worldwide registered trademark.
diff -ruN usb-step1/LICENCE.r8a779x_usb3 usb-step2/LICENCE.r8a779x_usb3
--- usb-step1/LICENCE.r8a779x_usb3	2025-11-18 03:34:08.614636949 +0000
+++ usb-step2/LICENCE.r8a779x_usb3	1970-01-01 00:00:00.000000000 +0000
@@ -1,26 +0,0 @@
-Copyright (c) 2014, Renesas Electronics Corporation
-All rights reserved.
-
-Redistribution and use in binary form, without modification, are permitted
-provided that the following conditions are met:
-
-1. Redistribution in binary form must reproduce the above copyright notice,
-   this list of conditions and the following disclaimer in the documentation
-   and/or other materials provided with the distribution.
-2. The name of Renesas Electronics Corporation may not be used to endorse or
-   promote products derived from this software without specific prior written
-   permission.
-3. Reverse engineering, decompilation, or disassembly of this software is
-   not permitted.
-
-THIS SOFTWARE IS PROVIDED "AS IS" AND RENESAS ELECTRONICS CORPORATION DISCLAIMS
-ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE, AND
-NONINFRINGEMENT OF THIRD PARTY RIGHTS. IN NO EVENT SHALL RENESAS ELECTRONICS
-CORPORATION BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
-OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
-SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-POSSIBILITY OF SUCH DAMAGE.
diff -ruN usb-step1/LICENCE.ueagle-atm4-firmware usb-step2/LICENCE.ueagle-atm4-firmware
--- usb-step1/LICENCE.ueagle-atm4-firmware	2025-11-18 03:34:08.615636951 +0000
+++ usb-step2/LICENCE.ueagle-atm4-firmware	1970-01-01 00:00:00.000000000 +0000
@@ -1,39 +0,0 @@
-This license applies to eagle4 firmware & DSPcode
-namely, the files eagleIV.fw  DSP4p.bin*
-
-| Copyright (2006) Ikanos Communications, Inc. 
-| 
-| Redistribution and use in source and binary forms, with or without 
-| modification, are permitted provided that the following 
-| conditions are met: 
-| 
-| * Redistribution of source code must retain the above copyright 
-| notice, this list of conditions and the following disclaimer. 
-| 
-| * Redistribution in binary form must reproduce the above 
-| copyright notice, this list of conditions and the following 
-| disclaimer in the documentation and/or other materials provided 
-| with the distribution. 
-| 
-| * The name of Ikanos Corporation may not be used to endorse         
-|   or promote products derived from this source code without specific     
-|   prior written consent of Ikanos Corporation. 
-| 
-| THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 
-| "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
-| LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR 
-| A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT 
-| OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
-| SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT 
-| LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, 
-| DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY 
-| THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
-| (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
-| OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
-| USER ACKNOWLEDGES AND AGREES THAT THE PURCHASE OR USE OF THIS SOFTWARE WILL 
-| NOT CREATE OR GIVE GROUNDS FOR A     
-|  LICENSE BY IMPLICATION, ESTOPPEL, OR OTHERWISE IN ANY INTELLECTUAL         
-|  PROPERTY RIGHTS (PATENT, COPYRIGHT, TRADE SECRET, MASK WORK, OR OTHER   
-|  PROPRIETARY RIGHT) EMBODIED IN ANY OTHER IKANOS HARDWARE OR SOFTWARE         
-|  EITHER SOLELY OR IN COMBINATION WITH THIS SOFTWARE. 
-
diff -ruN usb-step1/LICENCE.via_vt6656 usb-step2/LICENCE.via_vt6656
--- usb-step1/LICENCE.via_vt6656	2025-11-18 03:34:08.615636951 +0000
+++ usb-step2/LICENCE.via_vt6656	1970-01-01 00:00:00.000000000 +0000
@@ -1,25 +0,0 @@
-The following license applies to the binary-only VT6656 firmware
-as contained in the file "vntwusb.fw"
-================================================================
-Copyright 1998-2010 VIA Technologies, Inc. All Rights Reserved.
-
-Permission is hereby granted, free of charge, to any person
-obtaining a copy of this software and associated documentation
-files (the "Software"), to deal in the Software without
-restriction, including without limitation the rights to use,
-copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the
-Software is furnished to do so, subject to the following
-conditions:
-
-The above copyright notice and this permission notice shall be
-included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
-OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
-HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
-WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
-OTHER DEALINGS IN THE SOFTWARE.
diff -ruN usb-step1/WHENCE usb-step2/WHENCE
--- usb-step1/WHENCE	2025-11-18 03:34:08.805637382 +0000
+++ usb-step2/WHENCE	2025-11-18 03:34:30.595686769 +0000
@@ -17,19 +17,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: kaweth - USB KLSI KL5USB101-based Ethernet device
-
-File: kaweth/new_code.bin
-File: kaweth/new_code_fix.bin
-File: kaweth/trigger_code.bin
-File: kaweth/trigger_code_fix.bin
-
-Licence: Redistributable. See LICENCE.kaweth for details
-
-Found in hex form in the kernel source.
-
---------------------------------------------------------------------------
-
 Driver: dvb-ttusb-budget - Technotrend/Hauppauge Nova-USB devices
 
 File: ttusb-budget/dspbootcode.bin
@@ -40,123 +27,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: keyspan - USB Keyspan USA-xxx serial device
-
-File: keyspan/mpr.fw
-File: keyspan/usa18x.fw
-File: keyspan/usa19.fw
-File: keyspan/usa19qi.fw
-File: keyspan/usa19qw.fw
-File: keyspan/usa19w.fw
-File: keyspan/usa28.fw
-File: keyspan/usa28xa.fw
-File: keyspan/usa28xb.fw
-File: keyspan/usa28x.fw
-File: keyspan/usa49w.fw
-File: keyspan/usa49wlc.fw
-
-Converted from Intel HEX files, used in our binary representation of ihex.
-
-Licence: Redistributable. See LICENCE.keyspan for details
-
---------------------------------------------------------------------------
-
-Driver: keyspan_pda - USB Keyspan PDA single-port serial device
-
-File: keyspan_pda/keyspan_pda.fw
-Source: keyspan_pda/keyspan_pda.S
-
-File: keyspan_pda/xircom_pgs.fw
-Source: keyspan_pda/xircom_pgs.S
-
-Source: keyspan_pda/Makefile
-
-Licence: GPLv2 or later. See GPL-2 and GPL-3 for details.
-
-Compiled from original 8051 source into Intel HEX, used in our binary ihex form.
-
---------------------------------------------------------------------------
-
-Driver: emi26 - EMI 2|6 USB Audio interface
-
-File: emi26/bitstream.fw
-Version: 1.1.1.131
-Info: DATE=2001dec06
-
-File: emi26/firmware.fw
-Version: 1.0.2.916
-Info: DATE=12.02.2002
-
-File: emi26/loader.fw
-
-Converted from Intel HEX files, used in our binary representation of ihex.
-
-Licence: Redistributable. See LICENCE.emi26 for details
-
---------------------------------------------------------------------------
-
-Driver: emi62 - EMI 6|2m USB Audio interface
-
-File: emi62/bitstream.fw
-Version: 1.0.0.191
-Info: DATE= 2002oct28
-
-File: emi62/loader.fw
-Version: 1.0.2.002
-Info: DATE=10.01.2002
-
-File: emi62/midi.fw
-Version: 1.04.062
-Info: DATE=16.10.2002
-
-File: emi62/spdif.fw
-Version: 1.04.062
-Info: DATE=16.10.2002
-
-Converted from Intel HEX files, used in our binary representation of ihex.
-
-Licence: Unknown
-
---------------------------------------------------------------------------
-
-Driver: ti_usb_3410_5052 - USB TI 3410/5052 serial device
-
-File: ti_3410.fw
-Info: firmware 9/10/04 FW3410_Special_StartWdogOnStartPort
-
-File: ti_5052.fw
-Info: firmware 9/18/04
-
-Licence: Allegedly GPLv2+, but no source visible. Marked:
-	 Copyright (C) 2004 Texas Instruments
-
-Found in hex form in kernel source.
-
---------------------------------------------------------------------------
-
-Driver: ti_usb_3410_5052 - Multi-Tech USB cell modems
-
-File: mts_cdma.fw
-File: mts_gsm.fw
-File: mts_edge.fw
-
-Licence: Redistributable. See LICENCE.multitech for details
-
-Copyright (C) 2005 Multi-Tech Systems, Inc.
-
-Found in hex form in ftp://ftp.multitech.com/wireless/wireless_linux.zip
-
---------------------------------------------------------------------------
-
-Driver: ti_usb_3410_5052 - Multi-Tech USB fax modems
-
-File: mts_mt9234mu.fw
-File: mts_mt9234zba.fw
-
-Licence: Unknown
-
---------------------------------------------------------------------------
-
 Driver: whiteheat - USB ConnectTech WhiteHEAT serial device
 
 File: whiteheat.fw
@@ -2042,46 +1912,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: ueagle-atm - Driver for USB ADSL Modems based on Eagle IV Chipset
-
-File: ueagle-atm/CMV4p.bin.v2
-File: ueagle-atm/DSP4p.bin
-File: ueagle-atm/eagleIV.fw
-Version: 1.0
-
-Licence: Redistributable. See LICENCE.ueagle-atm4-firmware for details
-
---------------------------------------------------------------------------
-
-Driver: ueagle-atm - Driver for USB ADSL Modems based on Eagle I,II,III
-
-File: ueagle-atm/930-fpga.bin
-File: ueagle-atm/CMVeiWO.bin
-File: ueagle-atm/CMVepFR10.bin
-File: ueagle-atm/DSP9p.bin
-File: ueagle-atm/eagleIII.fw
-File: ueagle-atm/adi930.fw
-File: ueagle-atm/CMVep.bin
-File: ueagle-atm/CMVepFR.bin
-File: ueagle-atm/DSPei.bin
-File: ueagle-atm/CMV9i.bin
-File: ueagle-atm/CMVepES03.bin
-File: ueagle-atm/CMVepIT.bin
-File: ueagle-atm/DSPep.bin
-File: ueagle-atm/CMV9p.bin
-File: ueagle-atm/CMVepES.bin
-File: ueagle-atm/CMVepWO.bin
-File: ueagle-atm/eagleI.fw
-File: ueagle-atm/CMVei.bin
-File: ueagle-atm/CMVepFR04.bin
-File: ueagle-atm/DSP9i.bin
-File: ueagle-atm/eagleII.fw
-Version: 1.1
-
-Licence: Redistributable. See LICENCE.ueagle for details
-
---------------------------------------------------------------------------
-
 Driver: vxge - Exar X3100 Series 10GbE PCIe I/O Virtualized Server Adapter
 
 File: vxge/X3fw.ncf
@@ -2740,19 +2570,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: ene-ub6250 - ENE UB6250 SD card reader driver
-
-File: ene-ub6250/sd_init1.bin
-File: ene-ub6250/sd_init2.bin
-File: ene-ub6250/sd_rdwr.bin
-File: ene-ub6250/ms_init.bin
-File: ene-ub6250/msp_rdwr.bin
-File: ene-ub6250/ms_rdwr.bin
-
-Licence: Redistributable. See LICENCE.ene_firmware for details.
-
---------------------------------------------------------------------------
-
 Driver: s5p-mfc - Samsung MFC video encoder/decoder driver
 
 File: s5p-mfc.fw
@@ -3147,16 +2964,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: xhci-rcar - Renesas R-Car Gen2/3 USB 3.0 host controller driver
-
-File: r8a779x_usb3_v1.dlmem
-File: r8a779x_usb3_v2.dlmem
-File: r8a779x_usb3_v3.dlmem
-
-Licence: Redistributable. See LICENCE.r8a779x_usb3 for details.
-
---------------------------------------------------------------------------
-
 Driver: as102 - Abilis Systems Single DVB-T Receiver
 
 File: as102_data1_st.hex
